packages/react-native-bottom-tabs/ios/TabView/NewTabView.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+@available(iOS 18, macOS 15, visionOS 2, tvOS 18, *)
+struct NewTabView: AnyTabView {
+  @ObservedObject var props: TabViewProps
+  
+  var onLayout: (CGSize) -> Void
+  var onSelect: (String) -> Void
+  var updateTabBarAppearance: () -> Void
+
+  @ViewBuilder
+  var body: some View {
+    TabView(selection: $props.selectedPage) {
+      ForEach(props.children.indices, id: \.self) { index in
+        if let tabData = props.items[safe: index] {
+          let isFocused = props.selectedPage == tabData.key
+
+          if !tabData.hidden || isFocused {
+            let icon = props.icons[index]
+            let role: TabRole? = nil
```

Comment:
We can look for a `String` of `search` (To make this backwards-compatible <18.0) and then convert that into a `TabRole.search`


```suggestion
            var role: TabRole? {
              if tabData.tabRole == "search" {
                return TabRole.search
              }
              
              return nil
            }
```

Reply:
Of course we'd have to add the `tabRole` prop to `TabViewProvider` as well
```diff
diff --git a/packages/react-native-bottom-tabs/ios/TabViewProvider.swift b/packages/react-native-bottom-tabs/ios/TabViewProvider.swift
index 135c9a9..63669b5 100644
--- a/packages/react-native-bottom-tabs/ios/TabViewProvider.swift
+++ b/packages/react-native-bottom-tabs/ios/TabViewProvider.swift
@@ -13,7 +13,8 @@ public final class TabInfo: NSObject {
   public let activeTintColor: PlatformColor?
   public let hidden: Bool
   public let testID: String?
-
+  public let tabRole: String?
+  
   public init(
     key: String,
     title: String,
@@ -21,7 +22,8 @@ public final class TabInfo: NSObject {
     sfSymbol: String,
     activeTintColor: PlatformColor?,
     hidden: Bool,
-    testID: String?
+    testID: String?,
+    tabRole: String?
   ) {
     self.key = key
     self.title = title
@@ -30,6 +32,7 @@ public final class TabInfo: NSObject {
     self.activeTintColor = activeTintColor
     self.hidden = hidden
     self.testID = testID
+    self.tabRole = tabRole
     super.init()
   }
 }
@@ -267,7 +270,8 @@ public final class TabInfo: NSObject {
             sfSymbol: itemDict["sfSymbol"] as? String ?? "",
             activeTintColor: RCTConvert.uiColor(itemDict["activeTintColor"] as? NSNumber),
             hidden: itemDict["hidden"] as? Bool ?? false,
-            testID: itemDict["testID"] as? String ?? ""
+            testID: itemDict["testID"] as? String ?? "",
+            tabRole: itemDict["tabRole"] as? String ?? ""
           )
         )
       }
```

Reply:
makes sense to me, however i don't know if setting tabRole to a String? would be the right move, maybe using an Enum would make it more typesafe and future proof. Also, similarly to tabBarCustomization, it's probably better to add it in a follow up pr after this one is merged.

Reply:
Yeah `String?` isn't super ideal but the `TabRole` enum is only available in iOS18+ which would mean splitting into `NewTabViewProvider` and `LegacyTabViewProvider` as well as the split you already have.

There is currently only one role: .search but I think we could do `TabRole(rawValue: tabData.tabRole)` though which would still future proof it in case of future roles.

For sure can be done in a followup PR, it would obviously build upon yours and I couldn't figure out how to get GitHub to stack PRs

Reply:
you're right, that's probably the easiest way atm, and i guess we could implement the enum on the TS types (`type TabRole = 'search';`) so typescript users will have a typesafe way of adding a tab role. As for pr stacking i don't know either, we might have to wait for this one to be merged first

Reply:
I would remove passing role in this PR completely and create a separate PR implementing it. For PR stacking you can create a new branch off this PR and create a new PR pointing to this branch. 

Making it a string sounds good to me, we can use TypeScript to enforce users to pass `search` and if in the future Apple adds more roles we will be ready.